crates/sui-framework/packages/bridge/sources/treasury.move

```diff
@@ -77,4 +77,19 @@ module bridge::treasury {
             abort EUnsupportedTokenType
         }
     }
+
+    public fun token_decimals<T>(): u8 {
+        let coin_type = type_name::get<T>();
+        if (coin_type == type_name::get<BTC>()) {
+            8
+        } else if (coin_type == type_name::get<ETH>()) {
+            8
+        } else if (coin_type == type_name::get<USDC>()) {
+            6
+        } else if (coin_type == type_name::get<USDT>()) {
+            6
```

Comment:
can we get these from CoinMetadata somehow?

Reply:
CoinMetadata is shared after TreasuryCap creation, sadly it's not accessible after that unless we pass it in as an call arg....

Reply:
i see. But we can get consts from BTC.move etc?